sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -274,35 +300,48 @@ def convert_feature(fx):
                              'len(feature_names) = {0}, got {1}.'
                              .format(len(feature_names), i))
 
-    # compute averaged predictions
+    n_samples = X.shape[0]
+    if ((isinstance(subsample, int)
+         and (subsample >= n_samples or subsample <= 0))
+            or (isinstance(subsample, float)
+                and (subsample <= 0 or subsample >= 1))):
+        raise ValueError('subsample={0} should be either positive and smaller'
+                         ' than the number of samples {1} or a float in the '
+                         '(0, 1) range'.format(subsample, n_samples))
```

Comment:
I think we should accept larger values and just silently take all the samples instead of subsampling in that case.

Reply:
Fixed in https://github.com/scikit-learn/scikit-learn/pull/16619/commits/9a368799fa18ecea6f9f3df689c3141f1ea719b4